src/plugins/lua/arc.lua

```diff
@@ -639,7 +639,7 @@ local function prepare_arc_selector(task, sel)
       local ar_header = task:get_header('Authentication-Results')
 
       if ar_header then
-        local arc_match = string.match(ar_header, 'arc=(%w+)')
+        local arc_match = string.match(ar_header, '[; ]arc=(%w+)')
```

Comment:
It won't match a string where `arc=` is the first element...

Reply:
I don't see any possible choices to match that thing with Lua `match` I'm afraid. With PCRE it's trivial.

Reply:
aaah pants. I'll have another go at this soon

Reply:
Currently toying with the idea of the pattern matching below, which allows `arc=` to be detected regardless of where in the header value it appears and regardless of the prescence of `dmarc=`.

```
 s = "spf=softfail (sender IP is 3.212.108.9) \r\nsmtp.mailfrom=gmail.com;  arc=TESTING; dkim=fail (body hash did not verify)\r\n header.d=gmail.com;dmarc=fail action=none header.from=gmail.com;"
 for k, v in string.gmatch(s, "(%w+)=(%w+)") do
   if k == 'arc' then
    print(v)
   end
 end
```

However https://datatracker.ietf.org/doc/html/rfc7601#section-2.2 states CFWS may be present. Would rspamd have unfolded the headers during its parsing? Or is this something I need to factor in?

Reply:
Well, in fact that require an additional attention then. Rspamd provides headers unfolding for all headers but not SMTP comments removing if that means anything in this context.

Reply:
I've changed my MR based on my previous comment and I believe it is sufficiently better to consider merging:

- This may not be a total final solution (accounting for CFWS comments) but is superior to the current code which incorrectly matches `dmarc=`
- I have a few email corpus, in which I can see no examples of CFWS in problematic areas (i.e between `arc=pass`)

For example:
I cannot see an examples of:
```
Authentication-Results: example.com;
  dkim ( DKIM check) = pass (signature was verified) header.d=example.com;
  spf ( SPF check) = pass (sender IP is authorized) smtp.mailfrom=example.com;
  dmarc ( DMARC check) = pass (p=reject, dis=none) header.from=example.com;
  arc ( ARC check) = pass (instance=1) example.com
```
Which would not be detected in my, nor the original code.

but i see examples of:
```
Authentication-Results: example.com;
  dkim=pass (signature was verified) header.d=example.com;
  spf=pass (sender IP is authorized) smtp.mailfrom=example.com;
  dmarc=pass (p=reject, dis=none) header.from=example.com;
  arc=pass (i=1) example.com
```
Which would have incorrect behavior in the original code (matching `dmarc=`) but correct behavior in mine